pkg/controllers/node/node_controller.go

```diff
@@ -45,10 +45,15 @@ func (r *NodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		if errors.IsNotFound(err) {
 			log.Info("node not found", "req", req.NamespacedName)
 			deleted = true
+			if err := r.Service.SyncNodeStore(req.NamespacedName.Name, deleted); err != nil {
+				log.Error(err, "failed to sync node store", "req", req.NamespacedName)
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{}, nil
 		} else {
 			log.Error(err, "unable to fetch node", "req", req.NamespacedName)
+			return common.ResultNormal, client.IgnoreNotFound(err)
```

Comment:
Maybe just return err as in this else case client.IgnoreNotFound(err) will always be err?
```suggestion
			return common.ResultNormal, err
```

Or you can put it after the conditions.

```
if err := r.Client.Get(ctx, req.NamespacedName, node); err != nil {
		if errors.IsNotFound(err) {
			log.Info("node not found", "req", req.NamespacedName)
			deleted = true
			if err := r.Service.SyncNodeStore(req.NamespacedName.Name, deleted); err != nil {
				log.Error(err, "failed to sync node store", "req", req.NamespacedName)
				return common.ResultNormal, err
			}
		} else {
			log.Error(err, "unable to fetch node", "req", req.NamespacedName)
		}
                 return common.ResultNormal, client.IgnoreNotFound(err)
	}
```

Reply:
ok